app/src/main/java/me/hufman/androidautoidrive/music/spotify/SpotifyAuthStateManager.kt

```diff
@@ -96,12 +97,20 @@ class SpotifyAuthStateManager private constructor(val appSettings: MutableAppSet
 	}
 
 	/**
-	 * Updates the [AuthState] with the token response and authorization exception, replacing the old
-	 * AuthState in the shared preferences file with the new one.
+	 * Updates the [AuthState] with the provided [Token], replacing the old AuthState with the new one.
 	 */
-	fun updateTokenResponse(response: TokenResponse?, ex: AuthorizationException?) {
-		currentState.update(response, ex)
-		writeState(currentState)
+	fun updateTokenResponseWithToken(token: Token, clientId: String) {
```

Comment:
Ooo does this seem to help the spontaneous logouts? Feel free to pull this out to a separate PR too

Reply:
Sadly this doesn't, this was actually done for a couple of reasons, mainly the logic for updating the token response is more fitting to the SpotifyAuthStateManager, the SpotifyWebApi class shouldn't be concerning itself with building out the proper token response to update the AuthState with. There's also a long story with breaking Powermock 😄 so I was trying to pull logic out of the SpotifyWebApi class to see if I could avoid using Powermock on it so this is the first step